### PR TITLE
Query and update Pull request

### PR DESCRIPTION
**Pull request by HERMON-1995**
## In this project, I used transactions to update and delete records. The following are the changes:

- Inserted a data to the row table.
- Added species column.
- Wrote queries to answer the required questions.

## Here are the attached screenshots of the results of the queries.

- 
Set species to unspecified

- 
Roll back unspecified

- 
Species add digmon and pokemon

- 
Delete and roll back animals

- 
Update weight less than 0

- 
Number of animals

- 
Number of Animals no escape

- 
Average weight

- 
Neutered and max escape attempt

- 
Species and min weight

- 
Average and escape attempt

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -7,4 +7,37 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
...
+SELECT (id, name, species) from animals;
+ROLLBACK;
+SELECT (id, name, species) from animals;
+
+BEGIN;
+UPDATE animals SET species = 'digmon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+SELECT * FROM animals;
+COMMIT;
+
```
Comment (cynthiainga) [inline queries.sql]: - [ ] Good job so far but you should select species from animals to verify if the change persists after commit 🙃 🙃

File: queries.sql [hunk 2]
```diff
@@ -7,4 +7,37 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
...
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
```
Comment (cynthiainga) [inline queries.sql]: - [ ] After deleting from the animals table you should select all from the animal table to verify if the change persists after deleted 🙃 🙃

File: queries.sql [hunk 3]
```diff
@@ -7,4 +7,37 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
...
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+SELECT * FROM animals;
+COMMIT;
+
+SELECT COUNT(*) AS "Number of Animals" FROM animals;
+SELECT COUNT(*) AS "Number of Animals no escape" FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, MAX(escape_attempts) FROM animals GROUP BY neutered;
+SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY species;
+SELECT species, AVG(escape_attempts) FROM animals WHERE date_of_birth >= 'Jan 1, 1990' AND date_of_birth < 'Jan 1, 2000' GROUP BY species;
```
Comment (cynthiainga) [inline queries.sql]: - [ ] Good job so far but the date should be between 1990 and 2000 which means all the year 2000, you should fix the date should be `Dec 31, 2000` and the sign should be `<=`. Kindly fix that 🙃 🙃

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @HERMON-1995, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] Correct Gitflow ✔️ 
- [ ] Descriptive PR ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇 👇 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (HERMON-1995) [reply queries.sql]: Thank you I have added species selected. 😊

File: queries.sql (hunk 2 above)
Comment (HERMON-1995) [reply queries.sql]: Thank you I have selected all from the animal table. 😁

File: queries.sql (hunk 3 above)
Comment (HERMON-1995) [reply queries.sql]: Ohhh! I didn't notice that I have changed it. 😄

**Review by Ismailco — APPROVED** [review]
### Hi @HERMON-1995, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**